staging/src/k8s.io/kube-aggregator/pkg/cmd/server/start.go

```diff
@@ -142,16 +142,17 @@ func (o AggregatorOptions) RunAggregator(stopCh <-chan struct{}) error {
 		},
 	}
 
-	var err error
-	config.ExtraConfig.ProxyClientCert, err = ioutil.ReadFile(o.ProxyClientCertFile)
+	dynamicCert, err := dynamiccertificates.NewDynamicServingContentFromFiles("proxy-cert", o.ProxyClientCertFile, o.ProxyClientKeyFile)
 	if err != nil {
 		return err
 	}
-	config.ExtraConfig.ProxyClientKey, err = ioutil.ReadFile(o.ProxyClientKeyFile)
-	if err != nil {
+	if err := dynamicCert.RunOnce(); err != nil {
 		return err
 	}
 
+	go dynamicCert.Run(1, stopCh)
```

Comment:
pass `ProxyClientCertFile` and `ProxyClientKeyFile` to `ExtraConfig` and run the controller in `AddPostStartHookOrDie`

Reply:
oh yeah, that makes much more sense, thanks :)